ProjectProtocol/project-protocol-nextjs
# Tolgee import scripts

## 🛠️ Changes

Add a script to download tolgee locales and transpile them into the `src/locale/<language>` format expected by next-intl. This allows us to keep tolgee as a source of truth while using next-intl for localization.

Script runs automatically when starting the dev server. That means the server needs to be restarted to populate locale string changes from tolgee. Locale changes should be commited with your branch as these are static files within the nextjs project.

Also added a github action step to check that the locales are up-to date:
<img width="890" alt="image" src="https://github.com/ProjectProtocol/project-protocol-nextjs/assets/6488787/399caa1b-3dc2-49b4-b9c9-3f35561c0bce">


## 🧪 Testing

1. Make sure to add your tolgee API key to the .env.local as TOLGEE_ACCESS_TOKEN
2. Start the server, you should see:

```sh
> node ./bin/update-locales.js && next dev

UpdateLocales: ✅ Logged in as Tim Sandberg on app.tolgee.io for project Project Protocol (#7020). Welcome back!

UpdateLocales: Fetching strings from Tolgee...
UpdateLocales:    ✓ Success

UpdateLocales: Extracting strings...
UpdateLocales:    ✓ Success

UpdateLocales: ✅ Done!

FlattenLocalFiles: flattening account
FlattenLocalFiles: flattening agent
FlattenLocalFiles: flattening home
FlattenLocalFiles: flattening login
FlattenLocalFiles: flattening navigation
FlattenLocalFiles: flattening password_reset
FlattenLocalFiles: flattening rate_agent
FlattenLocalFiles: flattening rate_my_po
FlattenLocalFiles: flattening resources
FlattenLocalFiles: flattening shared
UpdateLocales: Cleaing up files...
UpdateLocales: Completed successfully!
 ⚠ Port 3000 is in use, trying 3001 instead.
 ⚠ Port 3001 is in use, trying 3002 instead.
  ▲ Next.js 14.2.4
  - Local:        http://localhost:3002
  - Environments: .env.local

 ✓ Starting...
 ✓ Ready in 1399ms
```
